### PR TITLE
WT-11062 Lock the page before checking the time aggregate

### PR DESCRIPTION
**Pull request by kommiharibabu**

### CODE REVIEW

**Comment by kommiharibabu** [comment]
@agorrod This is the lock change that I mentioned to protect against `ref->addr` being freed.

**Review by ajmorton — COMMENTED** [review]
One question about read/write vs spinlocks to start. Perf testing (linked on the Jira ticket) indicates things are ok but I'll investigate two outliers results to confirm .

File: src/include/btmem.h
Comment (kommiharibabu) [reply src/include/btmem.h:722]: The macros here are mainly used to hide the internal locking logic. Suppose we want to change from spinlock to rwlock, we don't have to change the caller's code. Changing the code here is enough.

File: src/include/btmem.h
Comment (kommiharibabu) [reply src/include/btmem.h:725]: Yes, I agree that `btcur_skip_page` needs read access on the page. It has already taken an exclusive lock on the ref, so changing it to `rwlock` may not help as it does not allow two cursors to check on the same ref anyway. I don't see any advantage in changing it to `rwlock`.

File: src/include/btree_inline.h
Comment (kommiharibabu) [reply src/include/btree_inline.h]: Thanks, fixed.

File: src/include/btmem.h
Comment (ajmorton) [reply src/include/btmem.h:725]: Thanks! All makes sense. Happy to leave the lock as is.

File: src/btree/bt_page.c
Comment (ajmorton) [inline src/btree/bt_page.c:66]: Error handling in this function is now spread across two locations, but the existing error handling being inside a switch statement was already unusual. I'll spin up a neweng ticket to clean this up outside of the PR.

**Review by ajmorton — APPROVED** [review]
LGTM, but I'd like to have a second pair of eyes on this before merging

**Review by agorrod — CHANGES_REQUESTED** [review]
Hari and I talked about the change this morning. I don't love adding in a lock acquisition to tree traversal.

He is going to try moving to safe-free semantics for the WT_REF::addr field, similarly to how split-safe free works in conjunction with generation tracking.

**Pull request closed without merging**
